### ISSUE TITLE
bug: clear called on _routeData.pendingChildren throws exception when it's immutable list

### ISSUE THREAD

**Issue by Octedia** [Milad-Akarie/auto_route_library#1641]
in routing_controller.dart line 697 I ran into a scenario where `_routeData.pendingChildren` is immutable and when clear is called it throws `UnsupportedError("Cannot clear an unmodifiable list");` when this happened the list is shown empty in debugger

I'm using the latest version 7.7.0 

I can't produce a sample but here is the stack

As you can see I use a`AutoTabsRouter` but it only happens when I set `reevaluateListenable` I'm using `ReevaluateListenable.stream`

it happens to me when the app first runs

```
The following UnsupportedError was thrown building TabContainerScreen(dependencies: [MyTheme, _InheritedTheme, _LocalizationsScope-[GlobalKey#84dd4]], state: _TabContainerScreenState#660d1):
Unsupported operation: Cannot clear an unmodifiable list

The relevant error-causing widget was: 
  TabContainerScreen TabContainerScreen:file:///Users/me/dev/my_markets/lib/app_router/app_router.gr.dart:80:22
When the exception was thrown, this was the stack: 
#0      UnmodifiableListMixin.clear (dart:_internal/list.dart:159:5)
#1      TabsRouter.setupRoutes (package:auto_route/src/router/controller/routing_controller.dart:697:32)
#2      _AutoTabsRouterIndexedStackState._setupController (package:auto_route/src/router/widgets/auto_tabs_router.dart:290:18)
#3      AutoTabsRouterState.didChangeDependencies (package:auto_route/src/router/widgets/auto_tabs_router.dart:200:7)
#4      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:5237:11)
#5      ComponentElement.mount (package:flutter/src/widgets/framework.dart:5062:5)
...     Normal element mounting (38 frames)
#43     Element.inflateWidget (package:flutter/src/widgets/framework.dart:3971:16)
#44     MultiChildRenderObjectElement.inflateWidget (package:flutter/src/widgets/framework.dart:6570:36)
#45     MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:6582:32)
...     Normal element mounting (178 frames)
#223    Element.inflateWidget (package:flutter/src/widgets/framework.dart:3971:16)
#224    MultiChildRenderObjectElement.inflateWidget (package:flutter/src/widgets/framework.dart:6570:36)
#225    MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:6582:32)
...     Normal element mounting (71 frames)
#296    Element.inflateWidget (package:flutter/src/widgets/framework.dart:3971:16)
#297    Element.updateChild (package:flutter/src/widgets/framework.dart:3702:20)
#298    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5111:16)
#299    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5251:11)
#300    Element.rebuild (package:flutter/src/widgets/framework.dart:4805:7)
#301    StatefulElement.update (package:flutter/src/widgets/framework.dart:5274:5)
#302    Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#303    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5111:16)
#304    Element.rebuild (package:flutter/src/widgets/framework.dart:4805:7)
#305    ProxyElement.update (package:flutter/src/widgets/framework.dart:5417:5)
#306    Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#307    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5111:16)
#308    Element.rebuild (package:flutter/src/widgets/framework.dart:4805:7)
#309    ProxyElement.update (package:flutter/src/widgets/framework.dart:5417:5)
#310    Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#311    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5111:16)
#312    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5251:11)
#313    Element.rebuild (package:flutter/src/widgets/framework.dart:4805:7)
#314    StatefulElement.update (package:flutter/src/widgets/framework.dart:5274:5)
#315    Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#316    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5111:16)
#317    Element.rebuild (package:flutter/src/widgets/framework.dart:4805:7)
#318    StatelessElement.update (package:flutter/src/widgets/framework.dart:5162:5)
#319    Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#320    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5111:16)
#321    Element.rebuild (package:flutter/src/widgets/framework.dart:4805:7)
#322    ProxyElement.update (package:flutter/src/widgets/framework.dart:5417:5)
#323    Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#324    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5111:16)
#325    Element.rebuild (package:flutter/src/widgets/framework.dart:4805:7)
#326    ProxyElement.update (package:flutter/src/widgets/framework.dart:5417:5)
#327    Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#328    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5111:16)
#329    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5251:11)
#330    Element.rebuild (package:flutter/src/widgets/framework.dart:4805:7)
#331    BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2780:19)
#332    WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:903:21)
#333    RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:358:5)
#334    SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1284:15)
#335    SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1214:9)
#336    SchedulerBinding.scheduleWarmUpFrame.<anonymous closure> (package:flutter/src/scheduler/binding.dart:939:7)
#340    _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:189:12)
(elided 3 frames from class _Timer and dart:async-patch)
```

**Comment by shawnk-kt** [comment]
Did you ever find a solution/workaround for this?

**Comment by Octedia** [comment]
@shawnk-kt I forked the package and placed a check if the list is empty before clear is called.

Before committing this feature make sure to test it well- it might not do what you expect

**Comment by RWolfing** [comment]
This should be fixed with `7.8.1`.
